backend/inspirehep/records/api/literature.py

```diff
@@ -157,18 +157,27 @@ def update(
 
     def add_files(self, data):
         if not current_app.config.get("FEATURE_FLAG_ENABLE_FILES", False):
+            LOGGER.info("Feature flag ``FEATURE_FLAG_ENABLE_FILES`` is disabled")
             return data
 
         if "deleted" in data and data["deleted"]:
+            LOGGER.info("Record is deleted", uuid=self.id)
             return data
 
         documents = data.pop("documents", [])
         figures = data.pop("figures", [])
 
+        self.pop("documents", None)
+        self.pop("figures", None)
+
         keys = []
         builder = LiteratureBuilder(record=data)
 
         for document in documents:
+            if document.get("hidden", False):
+                builder.add_document(**document)
+                continue
+
             file_data = self.add_file(document=True, **document)
             document.update(file_data)
             if "hidden" not in document:
```

Comment:
Now you do not need this `if` here right?

Reply:
removed